pkg/endpoint/policy.go

```diff
@@ -821,12 +821,19 @@ func (e *Endpoint) Regenerate(owner Owner, context *RegenerationContext) <-chan
 
 	go func(owner Owner, req *Request, e *Endpoint) {
 		var buildSuccess bool
+		defer func() {
+			// The external listener can ignore the channel so we need to
+			// make sure we don't block
+			select {
+			case req.ExternalDone <- buildSuccess:
+			default:
+			}
+			close(req.ExternalDone)
+		}()
 
 		err := e.RLockAlive()
 		if err != nil {
 			e.LogDisconnectedMutexAction(err, "before regeneration")
-			req.ExternalDone <- false
-			close(req.ExternalDone)
 			return
```

Comment:
The reason the `defer` is added is due to this return here when trying to acquire the endpoint lock, correct?

Reply:
yes